### PR TITLE
Add code for XMPP bots

### PR DESCRIPTION
**Pull request by rolandgeider**
This includes the code for both the collector and the output bots. Code is a tidied up version of the code from PRs 158 and 197, as well as the comments there.

Thanks to @helderfernandes1279 for the original code.

### CODE REVIEW

**Comment by sebix** [comment]
- The tests are failing because the BOTS-file should be sorted (as the ordering is directly used by the manager)
- What is the minimum version of sleekxmpp necessary (or tested with)? Please add this to the requirements
- As we are still supporting Python2, please add `from __future__ import unicode_literals` to the collector.
- The output bot class should be named `XMPPOutputBot` for the sake of consistent naming.
- Please don't import `sys` from `intelmq.lib.bot`

**Comment by rolandgeider** [comment]
@sebix I have fixed the issues in one go, while I'm all for atomic commits, I think in this case it's OK :)

**Comment by sebix** [comment]
Thanks for this contribution!

File: intelmq/bots/collectors/xmpp/collector.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        self.logger = logger
+        self.logger.info("XMPP connected")
+        self.send_message = send_message
+
+    def session_start(self, event):
+        self.send_presence()
+
+        try:
+            self.get_roster()
+        except IqError as err:
```
Comment (sebix) [inline intelmq/bots/collectors/xmpp/collector.py]: IqError is undefined.

File: intelmq/bots/collectors/xmpp/collector.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            self.logger.error('There was an error getting the roster')
+            self.logger.error(err.iq['error']['condition'])
+            self.disconnect()
+        except IqTimeout:
```
Comment (sebix) [inline intelmq/bots/collectors/xmpp/collector.py]: IqTimeout is undefinded.

File: intelmq/bots/collectors/xmpp/collector.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                                self.logger)
+            self.xmpp.connect(reattempt=True)
+            self.xmpp.process()
+
+    def killbot(self):
+        self.xmpp.disconnect(wait=True)
+        self.logger.info("Disconnected")
+
+
+class XMPPOutputBot(ClientXMPP):
```
Comment (sebix) [inline intelmq/bots/collectors/xmpp/collector.py]: XMPPOutputBot is not used

File: intelmq/bots/outputs/xmpp/output.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        self.logger = logger
+        self.logger.info("XMPP connected")
+        self.send_message = send_message
+
+    def session_start(self, event):
+        self.send_presence()
+
+        try:
+            self.get_roster()
+        except IqError as err:
```
Comment (sebix) [inline intelmq/bots/outputs/xmpp/output.py]: IqError is not defined

File: intelmq/bots/outputs/xmpp/output.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            self.logger.error('There was an error getting the roster')
+            self.logger.error(err.iq['error']['condition'])
+            self.disconnect()
+        except IqTimeout:
```
Comment (sebix) [inline intelmq/bots/outputs/xmpp/output.py]: IqTimeout is not defined

File: intelmq/bots/outputs/xmpp/output.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            self.logger.error('Server is taking too long to respond')
+            self.disconnect()
+
+    def message_logging(self, msg):
+
+        event = json.loads(msg['body'])
```
Comment (sebix) [inline intelmq/bots/outputs/xmpp/output.py]: json is unknown here

File: intelmq/bots/collectors/xmpp/collector.py (hunk 3 above)
Comment (sebix) [inline intelmq/bots/collectors/xmpp/collector.py]: XMPPOutputBot is not used

File: intelmq/bots/collectors/xmpp/collector.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+
+class XMPPCollectorBot(Bot):
+
+    xmpp = None
+
+    def process(self):
+
+        if self.xmpp is None:
+            self.xmpp = XMPPBot(self.parameters.xmpp_user + '@' + self.parameters.xmpp_server,
```
Comment (sebix) [inline intelmq/bots/collectors/xmpp/collector.py]: XMPPBot is undefined

File: intelmq/bots/BOTS [hunk 8]
```diff
@@ -578,6 +578,16 @@
                 "http_url": "http://vxvault.siri-urz.net/URL_List.php",
                 "rate_limit": 3600
             }
+        },
+        "XMPP collector": {
+            "description": "XMPP collector is the bot responsible to get events from xmpp server",
+            "module": "intelmq.bots.collectors.xmpp.collector",
+            "parameters": {
+                "rate_limit": "3600",
```
Comment (sebix) [inline intelmq/bots/BOTS]: needs to be an integer

File: intelmq/bots/collectors/xmpp/collector.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+from intelmq.lib.bot import Bot
+
+
+class XMPPCollectorBot(Bot):
+
+    xmpp = None
+
+    def process(self):
+
+        if self.xmpp is None:
```
Comment (sebix) [inline intelmq/bots/collectors/xmpp/collector.py]: Why not putting the connection establishing into init?

**Pull request merged**